src/api/app/controllers/webui/projects/project_configuration_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Webui
+  module Projects
+    class ProjectConfigurationController < WebuiController
+      before_action :set_project
+
+      def show
+        sliced_params = params.slice(:rev)
+        sliced_params.permit!
+        @content = @project.config.content(sliced_params.to_h)
+        switch_to_webui2
+        return if @content
+        flash[:error] = @project.config.errors.full_messages.to_sentence
+        redirect_to projects_path(nextstatus: 404)
+      end
+
+      def update
+        authorize @project, :update?
+        content = ::ProjectConfigurationService::ProjectConfigurationUpdater.new(@project, User.current, params).call
```

Comment:
I like the idea of using services, but in this case, I don't see the value of it, this could be a private method in `ProjectConfigurationController`

Reply:
the idea is to move away from private methods on controllers :)

Reply:
I also don't see much value in this particular service as there is not much logic into it and adds another layer of indirection. However, I'm fine with it leaving it like it is.

Reply:
IMO it will be easier to stub it to for instance, having a wrong or nil content.